doc/md/guides/deploying/flyio.mdx

```diff
@@ -0,0 +1,166 @@
+---
+id: fly.io
+slug: /guides/deploying/fly.io
+title: Using Atlas with Fly.io
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+[Fly.io](https://fly.io/) is a platform for running full stack apps and databases close to the users. Under the hood, Fly
+converts Docker images (OCI images) to [Firecracker microVMs](https://firecracker-microvm.github.io/). Fly allows
+the deployment of the app in different regions and route the requests based on the [app load and user closeness](https://fly.io/docs/reference/load-balancing/).
+
+Apps on Fly can be [deployed in one of three ways](https://fly.io/docs/reference/builders/): using a Dockerfile, a
+Docker image or a [buildpack](https://buildpacks.io/).
+
+In this guide, we will demonstrate how Atlas can be used to perform database schema migrations for Fly.io deployments process using a Dockerfile.
+We will assume that you have already have a Fly project and are able to deploy it, if you are new to Fly, [check the getting started guide](https://fly.io/docs/speedrun/).
+
+## Release command
+
+When you configure your Fly project, you can define a _release command_. This command is executed during the release phase, before the new version of your application is deployed.  If it fails, i.e exits with a status other than zero, Fly marks the deployment as failed mas well. The release command is the
+[recommended way](https://fly.io/docs/reference/configuration/#the-deploy-section) to run database migrations on Fly.
+
+:::note
+The release command only allows executing commands that are present in the application image, meaning that we
+have to embed the Atlas binary with our application. [We usually suggest a separate step for handling the
+migration](https://atlasgo.io/guides/deploying/intro#running-migrations-as-part-of-deployment-pipelines),
+but since this is a requirement from Fly, we don't have an option.
+:::
+
+## Defining the Dockerfile
+
+Using Docker multi-stage builds, we can create an easy-to-use Dockerfile. In this guide we will use a Go app as an example.
+Suppose our project structure is similar to the one below:
+
+```text
+.
+├── fly.toml
+├── go.mod
+├── go.sum
+├── main.go
+└── migrations
+    ├── 20221220000101_create_users.sql
+    └── atlas.sum
+```
+
+Our objective is to build an image that contains the Atlas binary, the database migrations and our application code. For
+our Go app the Dockerfile can be defined as:
+
+```dockerfile title="Dockerfile"
+FROM arigaio/atlas:latest as atlas
+
+# app stage
+FROM golang:1.19.2-bullseye as build
+WORKDIR /build
+ADD go.mod /build/go.mod
+ADD go.sum /build/go.sum
+ADD main.go /build/main.go
+RUN go build -o app main.go
+
+# final stage
+FROM ubuntu
```

Comment:
ubuntu is a fairly heavy image and contains lots of unrequired stuff. can we use a distroless base image? 
you can include busybox in there or similar if needed.

Reply:
or at least a slim alpine image?

Reply:
It's possible with `arigaio/atlas:latest-alpine`

Since we require `sh` `distroless` alone won't work, using the [debug](https://github.com/GoogleContainerTools/distroless#debug-images) version (that adds `busybox`) is possible and works (with a minor `Dockerfile` change)
```
FROM gcr.io/distroless/base-debian11:debug
COPY --from=atlas /atlas /atlas
COPY migrations /migrations
COPY --from=build /build/app /app
ENTRYPOINT [] # <-- required
CMD ["/app"]
```
We get a warning on the logs, but everything works.
```
could not setup swap: could not find `mkswap`, you need to install the correct util package for your distribution
 ```
 
I choose to use alpine.
- `alpine`:  81MB
- `distroless`: 95MB

Should we explain this on the guide?

Reply:
No it's fine. As long as we have it some thought 😉